eth/downloader/downloader.go

```diff
@@ -382,207 +395,270 @@ func (d *Downloader) Terminate() {
 	d.cancel()
 }
 
-// fetchHashes60 starts retrieving hashes backwards from a specific peer and hash,
-// up until it finds a common ancestor. If the source peer times out, alternative
-// ones are tried for continuation.
-func (d *Downloader) fetchHashes60(p *peer, h common.Hash) error {
-	var (
-		start  = time.Now()
-		active = p             // active peer will help determine the current active peer
-		head   = common.Hash{} // common and last hash
-
-		timeout     = time.NewTimer(0)                // timer to dump a non-responsive active peer
-		attempted   = make(map[string]bool)           // attempted peers will help with retries
-		crossTicker = time.NewTicker(crossCheckCycle) // ticker to periodically check expired cross checks
-	)
-	defer crossTicker.Stop()
-	defer timeout.Stop()
-
-	glog.V(logger.Debug).Infof("Downloading hashes (%x) from %s", h[:4], p.id)
-	<-timeout.C // timeout channel should be initially empty.
+// findAncestor61 tries to locate the common ancestor block of the local chain and
+// a remote peers blockchain. In the general case when our node was in sync and
+// on the correct chain, checking the top N blocks should already get us a match.
+// In the rare scenario when we ended up on a long soft fork (i.e. none of the
```

Comment:
"soft fork" is usually meant to refer to consensus protocol change, better name here is "long re-org"